### ISSUE TITLE
Unable to communicate with EC2 Metadata service to infer region

### ISSUE THREAD

**Issue by bronislav** [aws/amazon-ecs-agent#140]
I can't get ECS agent working cause it always crashes with "Unable to communicate with EC2 Metadata service to infer region".

In the same time curling meta-data endpoint working and I'm able to see region in the returning data.

I'm actually using CoreOS on my host. Below is the excerpts from the journal:
```
Jul 21 13:43:25 ip-10-5-4-120.empire systemd[1]: Started Amazon ECS Agent.
Jul 21 13:43:26 ip-10-5-4-120.empire docker[1611]: 2015-07-21T13:43:26Z [CRITICAL] Unable to communicate with EC2 Metadata service to infer region: Get http://169.254.169.254/2014-02-25/dynamic/instance-i
Jul 21 13:43:26 ip-10-5-4-120.empire docker[1611]: 2015-07-21T13:43:26Z [CRITICAL] Configuration key not set module="config" key="AWSRegion"
Jul 21 13:43:26 ip-10-5-4-120.empire docker[1611]: 2015-07-21T13:43:26Z [CRITICAL] Error loading config: Missing required fields: AWSRegion
Jul 21 13:43:26 ip-10-5-4-120.empire systemd[1]: amazon-ecs-agent.service: Main process exited, code=exited, status=1/FAILURE
Jul 21 13:43:26 ip-10-5-4-120.empire docker[1657]: ecs-agent
Jul 21 13:43:26 ip-10-5-4-120.empire systemd[1]: amazon-ecs-agent.service: Unit entered failed state.
Jul 21 13:43:26 ip-10-5-4-120.empire systemd[1]: amazon-ecs-agent.service: Failed with result 'exit-code'.
```
The full journal log for this period of time:
```
Jul 21 13:53:37 ip-10-5-4-120.empire systemd[1]: Starting Amazon ECS Agent...
Jul 21 13:53:37 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:37.579880450Z" level=info msg="POST /v1.19/containers/ecs-agent/kill?signal=KILL"
Jul 21 13:53:37 ip-10-5-4-120.empire docker[1762]: ecs-agent
Jul 21 13:53:37 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:37.609175032Z" level=info msg="DELETE /v1.19/containers/ecs-agent"
Jul 21 13:53:37 ip-10-5-4-120.empire docker[1768]: ecs-agent
Jul 21 13:53:37 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:37.643876950Z" level=info msg="POST /v1.19/images/create?fromImage=amazon%2Famazon-ecs-agent%3Alatest"
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: latest: Pulling from amazon/amazon-ecs-agent
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: f2a96ff1e4d6: Already exists
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: c84a1f04e8c9: Already exists
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: 045937459d59: Already exists
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: ac5487ac509d: Already exists
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: cfefd15f4b32: Already exists
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: cfefd15f4b32: Already exists
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: Digest: sha256:2634fc6cbdd56c786db548ebf0d1a74da5b2938f7f16d72420cd4fbc0171bb53
Jul 21 13:53:38 ip-10-5-4-120.empire docker[1775]: Status: Image is up to date for amazon/amazon-ecs-agent:latest
Jul 21 13:53:38 ip-10-5-4-120.empire systemd[1]: Started Amazon ECS Agent.
Jul 21 13:53:38 ip-10-5-4-120.empire polkitd[551]: Unregistered Authentication Agent for unix-process:1757:494371 (system bus name :1.64, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8) (disconnected from bus)
Jul 21 13:53:38 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:38.155415971Z" level=info msg="POST /v1.19/containers/create?name=ecs-agent"
Jul 21 13:53:38 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:38.176264335Z" level=info msg="POST /v1.19/containers/7c5bf337fc1eacdc3a8f3e768c9cda1952bf100dc14fc66a568450bce6a77315/attach?stderr=1&stdout=1&stream=1"
Jul 21 13:53:38 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:38.177169288Z" level=info msg="POST /v1.19/containers/7c5bf337fc1eacdc3a8f3e768c9cda1952bf100dc14fc66a568450bce6a77315/start"
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-timesyncd[497]: Network configuration changed, trying to establish connection.
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-udevd[1789]: Could not generate persistent MAC address for veth957448d: No such file or directory
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-udevd[1790]: Could not generate persistent MAC address for vethb512133: No such file or directory
Jul 21 13:53:38 ip-10-5-4-120.empire audit: ANOM_PROMISCUOUS dev=vethb512133 prom=256 old_prom=0 auid=4294967295 uid=0 gid=0 ses=4294967295
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: device vethb512133 entered promiscuous mode
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit_printk_skb: 54 callbacks suppressed
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1700 audit(1437486818.182:44): dev=vethb512133 prom=256 old_prom=0 auid=4294967295 uid=0 gid=0 ses=4294967295
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1300 audit(1437486818.182:44): arch=c000003e syscall=16 success=yes exit=0 a0=f a1=89a2 a2=c208578eec a3=0 items=0 ppid=1 pid=799 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="docker" exe="/usr/bin/docker" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1327 audit(1437486818.182:44): proctitle=646F636B6572002D2D6461656D6F6E002D2D686F73743D66643A2F2F
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: IPv6: ADDRCONF(NETDEV_UP): veth957448d: link is not ready
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: IPv6: ADDRCONF(NETDEV_CHANGE): veth957448d: link becomes ready
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Gained carrier
Jul 21 13:53:38 ip-10-5-4-120.empire audit[799]: SYSCALL arch=c000003e syscall=16 success=yes exit=0 a0=f a1=89a2 a2=c208578eec a3=0 items=0 ppid=1 pid=799 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="docker" exe="/usr/bin/docker" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire audit: PROCTITLE proctitle=646F636B6572002D2D6461656D6F6E002D2D686F73743D66643A2F2F
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Configured
Jul 21 13:53:38 ip-10-5-4-120.empire audit: NETFILTER_CFG table=nat family=2 entries=10
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1325 audit(1437486818.193:45): table=nat family=2 entries=10
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1300 audit(1437486818.193:45): arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7f99bb482390 items=0 ppid=621 pid=1793 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1327 audit(1437486818.193:45): proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D74006E6174002D4100444F434B4552002D7000746370002D64003132372E302E302E31002D2D64706F7274003531363738002D6A00444E4154002D2D746F2D64657374696E6174696F6E003137322E31372E302E353A35313637380000002D6900646F636B
Jul 21 13:53:38 ip-10-5-4-120.empire audit[1793]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7f99bb482390 items=0 ppid=621 pid=1793 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire audit: PROCTITLE proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D74006E6174002D4100444F434B4552002D7000746370002D64003132372E302E302E31002D2D64706F7274003531363738002D6A00444E4154002D2D746F2D64657374696E6174696F6E003137322E31372E302E353A35313637380000002D6900646F636B
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-networkd[523]: vethb512133: Configured
Jul 21 13:53:38 ip-10-5-4-120.empire audit: NETFILTER_CFG table=filter family=2 entries=10
Jul 21 13:53:38 ip-10-5-4-120.empire audit[1795]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7f1fe3e443f0 items=0 ppid=621 pid=1795 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire audit: PROCTITLE proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D740066696C746572002D4100444F434B45520000002D6900646F636B657230002D6F00646F636B657230002D7000746370002D64003137322E31372E302E35002D2D64706F7274003531363738002D6A00414343455054
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1325 audit(1437486818.202:46): table=filter family=2 entries=10
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1300 audit(1437486818.202:46): arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7f1fe3e443f0 items=0 ppid=621 pid=1795 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1327 audit(1437486818.202:46): proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D740066696C746572002D4100444F434B45520000002D6900646F636B657230002D6F00646F636B657230002D7000746370002D64003137322E31372E302E35002D2D64706F7274003531363738002D6A00414343455054
Jul 21 13:53:38 ip-10-5-4-120.empire audit: NETFILTER_CFG table=nat family=2 entries=11
Jul 21 13:53:38 ip-10-5-4-120.empire audit[1796]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7f2879e0d570 items=0 ppid=621 pid=1796 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire audit: PROCTITLE proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D74006E6174002D4100504F5354524F5554494E47002D7000746370002D73003137322E31372E302E35002D64003137322E31372E302E35002D2D64706F7274003531363738002D6A004D415351554552414445
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: audit: type=1325 audit(1437486818.207:47): table=nat family=2 entries=11
Jul 21 13:53:38 ip-10-5-4-120.empire audit: NETFILTER_CFG table=filter family=2 entries=0
Jul 21 13:53:38 ip-10-5-4-120.empire audit: NETFILTER_CFG table=nat family=2 entries=0
Jul 21 13:53:38 ip-10-5-4-120.empire audit[799]: SYSCALL arch=c000003e syscall=56 success=yes exit=1802 a0=40000011 a1=0 a2=0 a3=0 items=0 ppid=1 pid=799 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="docker" exe="/usr/bin/docker" key=(null)
Jul 21 13:53:38 ip-10-5-4-120.empire audit: PROCTITLE proctitle=646F636B6572002D2D6461656D6F6E002D2D686F73743D66643A2F2F
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-timesyncd[497]: Synchronized to time server 65.19.178.219:123 (2.coreos.pool.ntp.org).
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Lost carrier
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-timesyncd[497]: Network configuration changed, trying to establish connection.
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Removing non-existent address: fe80::42:acff:fe11:5/64 (valid for ever)
Jul 21 13:53:38 ip-10-5-4-120.empire kernel: eth0: renamed from veth957448d
Jul 21 13:53:38 ip-10-5-4-120.empire systemd[1]: Started docker container 7c5bf337fc1eacdc3a8f3e768c9cda1952bf100dc14fc66a568450bce6a77315.
Jul 21 13:53:38 ip-10-5-4-120.empire systemd[1]: Starting docker container 7c5bf337fc1eacdc3a8f3e768c9cda1952bf100dc14fc66a568450bce6a77315.
Jul 21 13:53:38 ip-10-5-4-120.empire systemd-timesyncd[497]: Synchronized to time server 65.19.178.219:123 (2.coreos.pool.ntp.org).
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-networkd[523]: vethb512133: Gained carrier
Jul 21 13:53:39 ip-10-5-4-120.empire kernel: docker0: port 2(vethb512133) entered forwarding state
Jul 21 13:53:39 ip-10-5-4-120.empire kernel: docker0: port 2(vethb512133) entered forwarding state
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-timesyncd[497]: Network configuration changed, trying to establish connection.
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-timesyncd[497]: Synchronized to time server 65.19.178.219:123 (2.coreos.pool.ntp.org).
Jul 21 13:53:39 ip-10-5-4-120.empire docker[1782]: 2015-07-21T13:53:39Z [CRITICAL] Unable to communicate with EC2 Metadata service to infer region: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: i/o timeout module="config"
Jul 21 13:53:39 ip-10-5-4-120.empire docker[1782]: 2015-07-21T13:53:39Z [CRITICAL] Configuration key not set module="config" key="AWSRegion"
Jul 21 13:53:39 ip-10-5-4-120.empire docker[1782]: 2015-07-21T13:53:39Z [CRITICAL] Error loading config: Missing required fields: AWSRegion
Jul 21 13:53:39 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:39.350605194Z" level=info msg="POST /v1.19/containers/7c5bf337fc1eacdc3a8f3e768c9cda1952bf100dc14fc66a568450bce6a77315/wait"
Jul 21 13:53:39 ip-10-5-4-120.empire kernel: veth957448d: renamed from eth0
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-timesyncd[497]: Network configuration changed, trying to establish connection.
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Configured
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Gained carrier
Jul 21 13:53:39 ip-10-5-4-120.empire audit: NETFILTER_CFG table=nat family=2 entries=12
Jul 21 13:53:39 ip-10-5-4-120.empire audit[1822]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7ff27754b540 items=0 ppid=621 pid=1822 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:39 ip-10-5-4-120.empire audit: PROCTITLE proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D74006E6174002D4400444F434B4552002D7000746370002D64003132372E302E302E31002D2D64706F7274003531363738002D6A00444E4154002D2D746F2D64657374696E6174696F6E003137322E31372E302E353A35313637380000002D6900646F636B
Jul 21 13:53:39 ip-10-5-4-120.empire audit: NETFILTER_CFG table=filter family=2 entries=11
Jul 21 13:53:39 ip-10-5-4-120.empire audit[1823]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7fb0ab3c63a0 items=0 ppid=621 pid=1823 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:39 ip-10-5-4-120.empire audit: PROCTITLE proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D740066696C746572002D4400444F434B45520000002D6900646F636B657230002D6F00646F636B657230002D7000746370002D64003137322E31372E302E35002D2D64706F7274003531363738002D6A00414343455054
Jul 21 13:53:39 ip-10-5-4-120.empire audit: NETFILTER_CFG table=nat family=2 entries=11
Jul 21 13:53:39 ip-10-5-4-120.empire audit[1824]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=4 a1=0 a2=40 a3=7f611b382450 items=0 ppid=621 pid=1824 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-multi" key=(null)
Jul 21 13:53:39 ip-10-5-4-120.empire audit: PROCTITLE proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D74006E6174002D4400504F5354524F5554494E47002D7000746370002D73003137322E31372E302E35002D64003137322E31372E302E35002D2D64706F7274003531363738002D6A004D415351554552414445
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Lost carrier
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-networkd[523]: veth957448d: Removing non-existent address: fe80::42:acff:fe11:5/64 (valid for ever)
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-networkd[523]: vethb512133: Lost carrier
Jul 21 13:53:39 ip-10-5-4-120.empire kernel: docker0: port 2(vethb512133) entered disabled state
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-networkd[523]: vethb512133: Removing non-existent address: fe80::c447:4fff:fe16:cbb1/64 (valid for ever)
Jul 21 13:53:39 ip-10-5-4-120.empire audit: ANOM_PROMISCUOUS dev=vethb512133 prom=0 old_prom=256 auid=4294967295 uid=0 gid=0 ses=4294967295
Jul 21 13:53:39 ip-10-5-4-120.empire kernel: device vethb512133 left promiscuous mode
Jul 21 13:53:39 ip-10-5-4-120.empire kernel: docker0: port 2(vethb512133) entered disabled state
Jul 21 13:53:39 ip-10-5-4-120.empire audit[800]: SYSCALL arch=c000003e syscall=44 success=yes exit=32 a0=f a1=c2083d20c0 a2=20 a3=0 items=0 ppid=1 pid=800 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="docker" exe="/usr/bin/docker" key=(null)
Jul 21 13:53:39 ip-10-5-4-120.empire audit: PROCTITLE proctitle=646F636B6572002D2D6461656D6F6E002D2D686F73743D66643A2F2F
Jul 21 13:53:39 ip-10-5-4-120.empire systemd[1]: amazon-ecs-agent.service: Main process exited, code=exited, status=1/FAILURE
Jul 21 13:53:39 ip-10-5-4-120.empire systemd-timesyncd[497]: Synchronized to time server 65.19.178.219:123 (2.coreos.pool.ntp.org).
Jul 21 13:53:39 ip-10-5-4-120.empire dockerd[621]: time="2015-07-21T13:53:39.438922772Z" level=info msg="POST /v1.19/containers/ecs-agent/stop?t=10"
Jul 21 13:53:39 ip-10-5-4-120.empire docker[1831]: ecs-agent
Jul 21 13:53:39 ip-10-5-4-120.empire systemd[1]: amazon-ecs-agent.service: Unit entered failed state.
Jul 21 13:53:39 ip-10-5-4-120.empire systemd[1]: amazon-ecs-agent.service: Failed with result 'exit-code'.
```

**Comment by janlisse** [comment]
Experiencing the same issue with latest agent (1.3.0). We are unable to add new nodes to our ECS cluster because of this issue right now.

**Comment by euank** [comment]
Thanks for reporting the issue. I'm able to intermittently reproduce this.

With the current dev branch, after merging #137, I haven't had any failures, so I'm marking this as pending-release for that change.

However, I was also able to workaround this issue with the existing agent by adding the following parameters to the agent's systemd unit:
```
       Restart=on-failure
       RestartPreventExitStatus=5
```

Best,
Euan

**Comment by janlisse** [comment]
Hi Euan,

when using the mentioned workaround where is the systemd unit file located?
When do you plan for the next release? Because this bug gives us a real headache right now.

Best regards,
Jan

**Comment by euank** [comment]
I created my unit-file via cloud-config, similar to how it's documented [here](https://coreos.com/os/docs/latest/booting-on-ecs.html). Where it's located depends on how you set it up; it should show up in `systemctl list-unit-files`.

I was also able to apply the workaround using `systemctl edit amazon-ecs-agent` and entering
```
[Service]
Restart=on-failure
RestartPreventExitStatus=5
```

**Comment by bronislav** [comment]
This workaround works for me as well.
Also, instead of editing actual unit file (for whatever reason) you may use systemd drop-ins.

**Comment by janlisse** [comment]
Just for the records, i am using the latest Amazon ECS optimized AMI. Not using Core OS.
The workaround described applies for Core OS only i guess?

**Comment by euank** [comment]
On the ECS Optimized AMI this actually should already be the default behaviour, so the workaround shouldn't be needed.

Are you not seeing it restart on failure? The logs located at `/var/log/ecs/ecs-init*` could also be helpful for checking if it restarted.

If it's not restarting or this issue is happening reliably even with restarts, that sounds like a subtly different issue that we'll have to look into.

**Comment by janlisse** [comment]
Ok i found the cause of this within one of our user-data scripts. It wasn't related at all to the ecs-agent. Sry for any inconveniences and many thanks for your help!

**Comment by samuelkarp** [comment]
We've released [v1.3.1](https://github.com/aws/amazon-ecs-agent/releases/tag/v1.3.1), which we believe has addressed this problem.  Please feel free to reopen or file a new issue if the problem persists.

Thanks!
Sam
